client/components/fixed-navigation-header/index.tsx

```diff
@@ -54,53 +54,31 @@ const ActionsContainer = styled.div`
 interface Props {
 	id?: string;
 	className?: string;
+	actions?: ReactNode;
 	children?: ReactNode;
 	navigationItems: { label: string; href?: string }[];
 	contentRef?: React.RefObject< HTMLElement >;
 	compactBreadcrumb?: boolean;
 }
 
 const FixedNavigationHeader: React.FunctionComponent< Props > = ( props ) => {
-	const { id, className, children, navigationItems, contentRef, compactBreadcrumb = false } = props;
-	const actionsRef = useRef< HTMLDivElement >( null );
-	const headerRef = useRef< HTMLElement >( null );
-
-	useEffect( () => {
-		if ( ! contentRef ) {
-			return;
-		}
-
-		const handleScroll = () => {
-			const headerHeight = headerRef?.current?.getBoundingClientRect().height;
-			const offset =
-				contentRef.current && headerHeight ? contentRef.current.offsetTop - headerHeight : 0;
-			const scrollPosition = window.scrollY;
-			const actionElement = actionsRef?.current;
-
-			if ( ! actionElement ) {
-				return;
-			}
-
-			if ( offset > 0 && scrollPosition < offset ) {
-				actionElement.style.visibility = 'hidden';
-			} else {
-				actionElement.style.visibility = 'visible';
-			}
-		};
-
-		handleScroll();
-
-		window.addEventListener( 'scroll', handleScroll );
-		return () => {
-			window.removeEventListener( 'scroll', handleScroll );
-		};
-	}, [ contentRef ] );
+	const {
+		id,
+		className,
+		children,
+		navigationItems,
+		contentRef,
+		compactBreadcrumb = false,
+		actions,
+	} = props;
+	const { headerRef, actionsRef } = useSticky( contentRef );
 
 	return (
 		<Header id={ id } className={ className } ref={ headerRef }>
 			<Container>
 				<Breadcrumb items={ navigationItems } compact={ compactBreadcrumb } />
-				<ActionsContainer ref={ actionsRef }>{ children }</ActionsContainer>
+				<ActionsContainer ref={ actionsRef }>{ actions }</ActionsContainer>
+				{ children }
```

Comment:
What's the rationale behind this change? This component was created to have two sections: 
- left - `Breadcrumb`
- right - `ActionsContainer` (as arbitrary children)

Do we really need a third section? Does unwrapping the `children` create any regressions?

Reply:
Also I tested the flow where we use the `FixedNavigationHeader` and no regressions or weird behavior where caused.